widget/graphic.go

```diff
@@ -84,19 +167,49 @@ func (g *Graphic) Render(screen *ebiten.Image) {
 func (g *Graphic) Update() {
 	g.init.Do()
 
+	if g.GIF != nil {
+		if g.gifCurrentImageAt.IsZero() {
+			g.gifCurrentImageAt = time.Now()
+		}
+
+		if time.Now().Sub(g.gifCurrentImageAt) > time.Duration(g.GIF.Delay[g.gifCurrentImage]*10)*time.Millisecond {
+			g.gifCurrentImageAt = time.Now()
+			g.gifCurrentImage += 1
+			if g.gifCurrentImage >= len(g.gifImages) {
+				g.gifCurrentImage = 0
+			}
+		}
+	}
+
 	g.widget.Update()
 }
 
 func (g *Graphic) draw(screen *ebiten.Image) {
-	if g.Image != nil {
-		opts := ebiten.DrawImageOptions{}
-		w, h := g.Image.Size()
-		opts.GeoM.Translate(float64((g.widget.Rect.Dx()-w)/2), float64((g.widget.Rect.Dy()-h)/2))
-		g.widget.drawImageOptions(&opts)
-		screen.DrawImage(g.Image, &opts)
-	} else if g.ImageNineSlice != nil {
+	if g.ImageNineSlice != nil {
 		g.ImageNineSlice.Draw(screen, g.widget.Rect.Dx(), g.widget.Rect.Dy(), g.widget.drawImageOptions)
+		return
+	}
+
+	if g.Image != nil {
+		g.Images = &GraphicImage{
+			Idle: g.Image,
+		}
+	} else if g.GIF != nil {
+		g.Images = &GraphicImage{
+			Idle: g.gifImages[g.gifCurrentImage],
+		}
+	}
+
+	i := g.Images.Idle
+	if g.widget.Disabled && g.Images.Disabled != nil {
+		i = g.Images.Disabled
 	}
+
+	opts := ebiten.DrawImageOptions{}
+	w, h := i.Size()
```

Comment:
This method is deprecated. Please update to the new suggested method  .Bounds().Size()